joy_teleop:
  ros__parameters:

    walk:
      type: topic
      interface_type: geometry_msgs/msg/TwistStamped
      topic_name: cmd_vel
      deadman_buttons: [4]
      #deadman_axes: [0]
      #deadman_axes_negative: [1]
      axis_mappings:
        linear-x:
          axis: 1
          scale: 0.5
          offset: -0.03
        angular-z:
          axis: 0
          scale: 0.5
          offset: 0
        linear-y:
          axis: 2
          scale: 0.3
          offset: 0
        linear-z:
          button: 2
          scale: 3.0
        header-frame_id:
          value: 'my_tf_frame'

    force_push:
      type: topic
      interface_type: geometry_msgs/msg/Wrench
      topic_name: base_link_wrench
      deadman_buttons: [5, 7]
      #deadman_axes: [0]
      #deadman_axes_negative: [1]
      axis_mappings:
        force-x:
          axis: 3
          scale: 40
          offset: 0
        force-y:
          axis: 2
          scale: 40
          offset: 0

    stop:
      type: topic
      interface_type: geometry_msgs/msg/Twist
      topic_name: cmd_vel
      deadman_buttons: [0, 2]
      #deadman_axes: [0]
      #deadman_axes_negative: [1]
      message_value:
        linear-x:
          value: 0.0
        angular-z:
          value: 0.0
        linear-y:
          value: 0.0

    hello:
      type: topic
      interface_type: std_msgs/msg/String
      topic_name: chatter
      deadman_buttons: [2]
      #deadman_axes: [0]
      #deadman_axes_negative: [1]
      message_value:
        data:
          value: 'Hello'

    array:
      type: topic
      interface_type: std_msgs/msg/UInt8MultiArray
      topic_name: bytes
      deadman_buttons: [5]
      #deadman_axes: [0]
      #deadman_axes_negative: [1]
      message_value:
        data:
          value: [1,3,3,7]

    array2:
      type: topic
      interface_type: std_msgs/msg/UInt8MultiArray
      topic_name: bytes
      deadman_buttons: [3]
      #deadman_axes: [0]
      #deadman_axes_negative: [1]
      message_value:
        data:
          value:
            - 4
            - 2

    array3:
      type: topic
      interface_type: std_msgs/msg/UInt8MultiArray
      topic_name: array3
      deadman_buttons: [0]
      #deadman_axes: [0]
      #deadman_axes_negative: [1]
      axis_mappings:
        data:
          index: 0
          axis: 0
          scale: 1
          offset: 0
        # leading dash are going to be striped away, but this allows us to have the same
        # field name (yaml doesn't allow duplicate keys)
        data-:
          index: 1
          axis: 1
          scale: 1
          offset: 0
        data--:
          index: 4
          axis: 2
          scale: 1
          offset: 0

    torso_up:
      type: action
      interface_type: teleop_tools_msgs/action/Increment
      action_name: torso_controller/increment
      action_goal:
        increment_by: [0.05]
      buttons: [4]
      # axes: [4]
      # axes_negative: [5]

    torso_up_axes:
      type: action
      interface_type: teleop_tools_msgs/action/Increment
      action_name: torso_controller/increment
      action_goal:
        increment_by: [0.05]
      buttons: [4]
      axes: [5]

    torso_down_axes_negative:
      type: action
      interface_type: teleop_tools_msgs/action/Increment
      action_name: torso_controller/increment
      action_goal:
        increment_by: [-0.05]
      buttons: [4]
      axes_negative: [5]

    torso_down:
      type: action
      interface_type: teleop_tools_msgs/action/Increment
      action_name: torso_controller/increment
      action_goal:
        increment_by: [-0.05]
      buttons: [6]
      # axes: [4]
      # axes_negative: [5]

    fibonacci:
      type: action
      interface_type: example_interfaces/action/Fibonacci
      action_name: fibonacci
      action_goal:
        order: 5
      buttons: [4, 5, 6, 7]
      # axes: [4]
      # axes_negative: [5]

    add_two_ints:
      type: service
      interface_type: example_interfaces/srv/AddTwoInts
      service_name: add_two_ints
      service_request:
        a: 11
        b: 31
      buttons: [10]
      # axes: [4]
      # axes_negative: [5]